tests/Unit/Configuration/Definition/ActionConfigDefinitionTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorCommon\Tests\Unit\Configuration\Definition;
+
+use Keboola\DbExtractorCommon\Configuration\BaseExtractorConfig;
+use Keboola\DbExtractorCommon\Configuration\Definition\ActionConfigDefinition;
+use Keboola\DbExtractorCommon\Tests\Unit\Configuration\__fixtures\ConfigParametersProvider;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+
+class ActionConfigDefinitionTest extends TestCase
+{
+    public function testConfigDatabaseNodeEmptyThrowsException(): void
+    {
+        $configuration = [
+            'parameters' => [
+                'db' => [],
+            ],
+        ];
+
+        $this->expectException(InvalidConfigurationException::class);
+        $this->expectExceptionMessage('The child node "user" at path "root.parameters.db" must be configured.');
+        new BaseExtractorConfig($configuration, new ActionConfigDefinition());
+    }
+
+    public function testConfigDatabaseNodeWithUserOnlyThrowsException(): void
+    {
+        $configuration = [
+            'parameters' => [
+                'db' => [
+                    'user' => 'username',
+                ],
+            ],
+        ];
+
+        $this->expectException(InvalidConfigurationException::class);
+        $this->expectExceptionMessage('The child node "#password" at path "root.parameters.db" must be configured.');
+        new BaseExtractorConfig($configuration, new ActionConfigDefinition());
+    }
+
+    public function testConfigDatabaseNodeWithMinimumParameters(): void
```

Comment:
Upravil bych na `testValidConfiguration(array $expectedConfiguration, array $inputConfiguration): void` a dataprovider. Ale hlavně teda to kontrolování výsledného configu, že je stejný.

Důvod je ten, že tenhle test by nezachytil změny chování, kdy to začne to vracet něco jiného. Třeba když je ssh.enabled=false, tak unset(ssh). Tenhle test by v pohodě prošel a přitom by se měl rozbít, aby tě upozornil, že je to backwards incompatible.

Reply:
pojmenovávání jde udělat pomocí klíčů toho dataprovider pole ;) JFYI

Reply:
Kvuli tomuto bych spise pridal test primo na tridu Config*. Tady bych chtel spise testovat (ne)validni konfiguraci a tim skoncit.

Reply:
Takže tam budeš pokrývat ty samé use-casy? To mi přijde zbytečný, ale jestli je ti to jedno...

Reply:
Nijak zvlast se to duplikovat nebude. Nehlede na to, ze ted jsem to cele predelal na 2 testy s data providery a tam by se ten config spatne testoval.

Reply:
Pridam testy v dalsim PR